apps/webapp/src/components/shared/select-token-modal.tsx

```diff
@@ -65,9 +64,8 @@ export default function SelectTokenModal({
             </div>
             <div className='flex flex-col gap-2'>
               {balances.map((b, i) => {
-                const index =
-                  b.address.addressView.case === 'decoded' &&
-                  b.address.addressView.value.index?.account;
+                const index = getAddressIndex(b.address).account;
```

Comment:
Note that this is a change in behavior — this component will now throw if the address index is undefined. Assuming that's intentional?